api/src/controllers/search.js

```diff
@@ -738,9 +738,13 @@ const generateMatchesForAggregation = async function (and, or, nor, searchProper
 };
 
 const executeQuery = async function (args, res, next) {
+  if (!args.swagger.params.dataset.value) {
+    defaultLog.info('Bad Request');
+    QueryActions.sendResponse(res, 400, {});
```

Comment:
Execution needs to stop here.  The search API still crashes if the code execution continues.

Reply:
997c247